src/pages/ImportData.js

```diff
@@ -77,40 +77,224 @@ const ResponseFragment = ({ title, snapshotResponses, responseIndex }) => {
   ])
 }
 
-const ImportData = () => {
+const ImportDataOverview = ({ header, snapshots, isDataset, snapshotResponses, url }) => div({ style: styles.card }, [
+  h2({ style: styles.title }, [header]),
+  !_.isEmpty(snapshots) ?
+    div({ style: { marginTop: 20, marginBottom: 60 } }, [
+      'Dataset(s):',
+      ul({ style: { listStyle: 'none', position: 'relative', marginLeft: 0, paddingLeft: '2rem' } }, [
+        _.flow(
+          Utils.toIndexPairs,
+          _.map(([mapindex, { title, id }]) => li({
+            key: `snapshot_${id}`,
+            style: {
+              fontSize: 16,
+              fontWeight: 'bold',
+              marginTop: 20,
+              paddingTop: mapindex ? 20 : 0,
+              borderTop: `${mapindex ? 1 : 0}px solid #AAA`
+            }
+          }, [
+            h(ResponseFragment, { snapshotResponses, responseIndex: mapindex, title })
+          ])))(snapshots)
+      ])
+    ]) :
+    url && div({ style: { fontSize: 16 } }, ['From: ', new URL(url).hostname]),
+  div({ style: { marginTop: '1rem' } }, [
+    `The ${isDataset ? 'dataset' : 'snapshot'}(s) you just chose to import to Terra will be made available to you `,
+    'within a workspace of your choice where you can then perform analysis.'
+  ])
+])
+
+// ImportDataDestination handles selecting which workspace to import to
+const ImportDataDestination = ({ workspaceId, templateWorkspaces, template, userHasBillingProjects, isDataset, authorizationDomain, onImport, isImporting }) => {
```

Comment:
This function seems to be doing a lot and requires many arguments, but I'm not sure how it could be broken apart into smaller more concise chunks.

Reply:
Great question. I think it will always have a lot of arguments, just because we need to pass that data down from the parent, but I think we could potentially abstract the individual cases into their own global level components. I wasn't sure whether the value there would outweigh needing to plumb through the props to all of them, as opposed to leaving them as methods within this component. (If we do pull them out globally, I would probably also pull out the default case here for consistency)